vmdk_plugin/utils/refcount/refcnt.go

```diff
@@ -272,25 +272,34 @@ func (r RefCountsMap) discoverAndSync(c *client.Client, d drivers.VolumeDriver)
 	filters.Add("status", "running")
 	filters.Add("status", "paused")
 	filters.Add("status", "restarting")
-	containers, err := c.ContainerList(context.Background(), types.ContainerListOptions{
+
+	ctx, cancel := context.WithTimeout(context.Background(), dockerConnTimeout*time.Second)
+	defer cancel()
+	containers, err := c.ContainerList(ctx, types.ContainerListOptions{
 		All:    true,
 		Filter: filters,
 	})
 	if err != nil {
+		log.Errorf("ContainerList failed (err: %v)", err)
 		return err
 	}
 
 	log.Debugf("Found %d running or paused containers", len(containers))
 	for _, ct := range containers {
-		containerJSONInfo, err := c.ContainerInspect(context.Background(), ct.ID)
+		ctx_inspect, cancel_inspect := context.WithTimeout(context.Background(), dockerConnTimeout*time.Second)
+		defer cancel_inspect()
+		containerJSONInfo, err := c.ContainerInspect(ctx_inspect, ct.ID)
```

Comment:
need to return (not continue) on error.
It would be usefule to mention in the comment that we delibertly leave the refcount table half-populated in this case, since it will improve the chances of correct operation after this specific failure to communicate to Docker